packages/offix-client/src/OfflineClient.ts

```diff
@@ -100,21 +101,27 @@ export class OfflineClient implements ListenerProvider {
    *
    * @param options the MutationHelperOptions to create the mutation
    */
-  public offlineMutation<T = any, TVariables = OperationVariables>(
+  public offlineMutate<T = any, TVariables = OperationVariables>(
     options: MutationHelperOptions<T, TVariables>): Promise<FetchResult<T>> {
     if (!this.apolloClient) {
       throw new Error("Apollo offline client not initialised before mutation called.");
+    }
+
+    const mutationOptions = createMutationOptions<T, TVariables>(options);
+    if (this.config.networkStatus.isOffline) {
+      // FIXME use internal part of the offline engine for this
+      OfflineMutationsHandler.markAsOffline(mutationOptions);
```

Comment:
I think this function should instead take the operation. We then use setContext on this operation. As it is the function takes a context object but we can't mutate `context` from `mutationOptions`.

Reply:
To be honest we should probably just use `setContext` directly from within this function.

Reply:
We are not in links. The operation was not created. Our helpers also work directly on the context because this is exactly before we passing that to mutate function. We can execute any operation on this object.

Reply:
Ah yes, my bad - Friday evening hiccup!